src/lib/utils/hooks/SynapseAPI/useEntity.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Hooks to access Entity Services in Synapse
+ */
+
+import {
+  useMutation,
+  UseMutationOptions,
+  useQuery,
+  useQueryClient,
+  UseQueryOptions,
+} from 'react-query'
+import { SynapseClient } from '../..'
+import { SynapseClientError } from '../../SynapseClient'
+import { useSynapseContext } from '../../SynapseContext'
+import { EntityJson } from '../../synapseTypes'
+
+export function useGetJson(
+  entityId: string,
+  options?: UseQueryOptions<EntityJson, SynapseClientError>,
+) {
+  const { accessToken } = useSynapseContext()
+  return useQuery<EntityJson, SynapseClientError>(
+    [accessToken, 'entity', entityId, 'json'],
+    () => SynapseClient.getEntityJson(entityId, accessToken),
+    options,
+  )
+}
+
+export function useUpdateViaJson(
+  entityId: string,
+  json: unknown,
+  options?: UseMutationOptions<EntityJson, SynapseClientError>,
+) {
+  const queryClient = useQueryClient()
+  const { accessToken } = useSynapseContext()
+
+  return useMutation<EntityJson, SynapseClientError>(
+    [accessToken, 'entity', entityId, 'json'],
+    () => SynapseClient.updateEntityJson(entityId, json, accessToken),
+    {
+      ...options,
+      onSuccess: async (data, variables, ctx) => {
+        await queryClient.invalidateQueries([accessToken, 'entity', entityId], {
+          exact: false,
+        })
+        queryClient.setQueryData(
+          [accessToken, 'entity', entityId, 'json'],
+          data,
+        )
+
+        if (options?.onSuccess) {
+          await options.onSuccess(data, variables, ctx)
+        }
+      },
+    },
+  )
```

Comment:
First time using `useMutation`! 

When the query succeeds, we invalidate all query cache data that starts with the key `[accessToken, 'entity', entityId]`. If we can organize our cache keys properly, this should blow away all entity data for that specific ID.

We take advantage of the fact that the response contains the updated entity (most importantly, the new eTag) by calling `setQueryData` and passing the same query key as the corresponding `GET` hook.